### PR TITLE
Add apple_connect welcome notification broadcast

### LINKED ISSUE #16059: users aren't getting onboarding broadcast messages
**Describe the bug**

We expect to send a sequence of welcome notifications to users during the first week after signup. This is failing after the initial follow up (participate in the welcome thread), and related to an error raised when checking for authentication providers.

The Broadcasts::WelcomeNotification::Generator checks to see which authentication methods a user has not enabled, and suggests the first one as an option.

For most users on DEV, this will be to sign in with an Apple id.

However, there's no Broadcast created with this title "Welcome Notification: apple_connect", and the generator raises a (captured) error about the record not being found, then stops, preventing both the authentication message, and all further messages from being sent.

**To Reproduce**

Set a user's created_at to 3.days.ago.
Ensure the user does not have apple identity configured, and that apple is an enabled identity provider.
Destroy all notifications (or all notifications for this user, or all notifications of type broadcast for this user)

Call Broadcasts::WelcomeNotification::Generator.call(user.id)

Observe that a notification is created for welcome_thread (as a sidekiq job)

Call the generator on this user again

Observe that no job is enqueued, no new notification is created. The last query was to load the (missing) apple_connect broadcast:

```
  Broadcast Load (0.3ms)  SELECT "broadcasts".* FROM "broadcasts" WHERE "broadcasts"."active" = $1 AND "broadcasts"."title" = $2 LIMIT $3  [["active", true], ["title", "Welcome Notification: apple_connect"], ["LIMIT", 1]]
```

**Expected behavior**

An auth notification is sent, and on subsequent days the remaining messages are sent (customize feed, customize experience, ask questions). 

**Screenshots**

**Additional context**

The solution to this issue likely involves both an update script to create the missing Broadcast message, as well as a modification to the way the error handling is performed in Generator#call (when any step in the flow raises an error, the process for this user is halted, and since each day the checks run in order, the next day the same error will be raised).

Ideally, an individual step raising an error would not block subsequent steps. Creating the message resolves the most common failure location.

### PR DESCRIPTION
**Pull request by djuber**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [ ] Feature
- [x] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

We expect there to be a Broadcast with title "Welcome Notification: apple_connect", but no such item exists.

## Related Tickets & Documents

fixes #16059 (in conjunction with #16060)

## QA Instructions, Screenshots, Recordings

Run the data update script. A broadcast should be created.

Run it again, nothing should happen.

### UI accessibility concerns?

None

## Added/updated tests?

- [x] Yes

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

- [x] I will share this change internally with the appropriate teams

### CODE REVIEW

**Comment by benhalpern** [comment]
Hmmm unless I'm misunderstanding, I think the result is that we don't want to _need_ to have Apple, and there are a lot of reasons not to... We just don't want to error out of the whole process if it _doesn't_ exist.

I don't think this needs to be a data update script as long as it's okay for this just to not exist, eh?

File: lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
+module DataUpdateScripts
+  class InsertAppleConnectBroadcastMessage
+    def run
+      title = "Welcome Notification: apple_connect"
+      return if Broadcast.find_by title: title
+
+      message = "You're on a roll! 🎉  Do you have an Apple account? " \
```
Comment (djuber) [inline lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb:7]: this text matches both the other (recent) connect options like forem or facebook, as well as the factory.

Question: should this also go into the seeds.rb? I think so.

**Comment by djuber** [comment]
> I don't think this needs to be a data update script as long as it's okay for this just to not exist, eh?

yes, the error handling in the other PR handles the case where this doesn't exist.

The existence of the broadcast is fine - it's only _sent_ if it's enabled (apple is an identity provider) and it's not used by the user (they're logged in with some other provider, but might want to add apple as well).

We're already filtering just just enabled (for the system) providers, and removing any enabled (for this user) identities, and suggesting one they don't have.

This isn't strictly necessary, but I think there's an implicit assumption that each auth provider also has a broadcast message. 

We _could_ limit or order the providers so apple was less likely to be first (put them in some order of popularity or internal ranking), but otherwise adding this message fixes a few things, and if apple is disabled it does no harm to add it to the database (and never send it).

**Review by joshpuetz — APPROVED** [review]

File: lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+                "Consider <a href='/settings'>connecting it</a>."
+
+      Broadcast.create!(
+        title: title,
+        processed_html: message,
+        type_of: "Welcome",
+        active: true,
+      )
```
Comment (jgaskins) [inline lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb:15]: If someone runs `rails db:seed data_updates:run`, will this fail or cause any other problems?

Comment (djuber) [reply lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb:15]: the early return (find by title and return silently if present) avoids that conflict.

**Review by juliannatetreault — APPROVED** [review]

File: lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb (hunk 2 above)
Comment (jgaskins) [reply lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb:15]: Ballin. 💯 Totally missed that.

Mic drop

**Review by jgaskins — APPROVED** [review]

File: lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb (hunk 2 above)
Comment (djuber) [reply lib/data_update_scripts/20220111192031_insert_apple_connect_broadcast_message.rb:15]:
```
 bundle exec rails db:drop db:create db:schema:load db:seed data_updates:run
```

This succeeded without error - and insert apple broadcast message was marked as finished in the logs :relieved:

**Pull request merged**